conda_smithy/linter/hints.py

```diff
@@ -291,24 +292,67 @@ def hint_noarch_python_use_python_min(
                 ):
                     break
             else:
-                hint += (
+                hint.add(
                     f"\n   - For the `{section_name}` section of the recipe, you should usually use `{report_syntax}` "
                     f"for the `python` entry."
                 )
+    return hint
 
-        if hint:
-            hint = (
-                (
-                    "`noarch: python` recipes should usually follow the syntax in "
-                    "our [documentation](https://conda-forge.org/docs/maintainer/knowledge_base/#noarch-python) "
-                    "for specifying the Python version."
-                )
-                + hint
-                + (
-                    "\n   - If the package requires a newer Python version than the currently supported minimum "
-                    "version on `conda-forge`, you can override the `python_min` variable by adding a "
-                    "Jinja2 `set` statement at the top of your recipe (or using an equivalent `context` "
-                    "variable for v1 recipes)."
+
+def hint_noarch_python_use_python_min(
+    host_reqs,
+    run_reqs,
+    test_reqs,
+    outputs_section,
+    noarch_value,
+    recipe_version,
+    hints,
+):
+    hint = set()
+
+    if outputs_section:
+        for output in outputs_section:
+            requirements = output.get("requirements", {})
+            if isinstance(requirements, Mapping):
+                output_host_reqs = requirements.get("host")
+                output_run_reqs = requirements.get("run")
+            else:
+                output_host_reqs = requirements
+                output_run_reqs = requirements
+
+            hint.update(
```

Comment:
update or add?

Reply:
Update. The `inner` function constructs a set of hints applying to a single output, then here we merge them into all hints applicable to a recipe. Not the most pretty solution, I admit, but I figured we don't want to repeat the same hint multiple times when it applies to multiple outputs.

Reply:
We do need to make it clear that a given hint applies to a given output. IIRC, In other parts of the linter we tolerate the repeats and report everything for each output. Alternatively we could aggregate the output names that each hint applies to and explicitly list them in each hint. I don't think the code as is will be clear enough.

Reply:
Okay, I'll update it tomorrow.

Reply:
I've added the output name to the message, when it applies to a specific output. Please let me know if this form is fine or do you prefer putting it there some other way.